File: DoitMission_09/Doit09_Java/app/src/main/java/com/example/doit09_java/MainActivity.java
package com.example.doit09_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    private CustomFragment customFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        customFragment = (CustomFragment) getSupportFragmentManager().findFragmentById(R.id.customfragment);
        customFragment = new CustomFragment();
    }
}

File: DoitMission_11/Doit11_Kotlin/app/src/main/java/com/example/doit11_kotlin/MainActivity.kt
package com.example.doit11_kotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    companion object {
        const val name = "NAME"
        const val message = "MESSAGE"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sendbutton.setOnClickListener {
            val serviceIntent = Intent(applicationContext,TextService::class.java)
            serviceIntent.putExtra(name,"Service")
            serviceIntent.putExtra(message,editText.text.toString())
            Log.e("Start","버튼 클릭!")
            startService(serviceIntent)
        }
        val intent1 = intent
        processIntent(intent1)
    }

    override fun onNewIntent(intent: Intent) {
        processIntent(intent)
        super.onNewIntent(intent)
    }

    private fun processIntent(intent : Intent){
        if(intent.getStringExtra(message) != null) {
            val processmessage = intent.getStringExtra(message)
            receiveText.text = processmessage
        }
    }
}
